src/bam/record.rs

```diff
@@ -867,7 +867,7 @@ impl<'a> ops::Index<usize> for Seq<'a> {
 unsafe impl<'a> Send for Seq<'a> {}
 unsafe impl<'a> Sync for Seq<'a> {}
 
-#[derive(PartialEq, Eq, Debug, Clone, Copy, Hash)]
+#[derive(PartialEq, PartialOrd, Eq, Debug, Clone, Copy, Hash, Deserialize, Serialize)]
```

Comment:
Yo need to add the serde derives in a separate line like this:
`#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]`

Reply:
I've reverted most of the changes but this does not seem to work just yet, as `Serialize` and `Deserialize` are out of scope.
Instead I had to define `serde_derive` as feature.
While this is compiling and all tests are passing serializing a Cigar still does not work returning `the trait `cli::_::_serde::Serialize` is not implemented for `rust_htslib::bam::record::Cigar``